installer/dev/install.cpp

```diff
@@ -66,9 +66,24 @@ namespace WindowsAppRuntimeInstaller {
             return true;
         }
 
-        SYSTEM_INFO systemInfo{};
-        GetNativeSystemInfo(&systemInfo);
-        const auto systemArchitecture{ static_cast<ProcessorArchitecture>(systemInfo.wProcessorArchitecture) };
+        USHORT processMachine{ IMAGE_FILE_MACHINE_UNKNOWN };
+        USHORT nativeMachine{ IMAGE_FILE_MACHINE_UNKNOWN };
+        THROW_IF_WIN32_BOOL_FALSE(::IsWow64Process2(::GetCurrentProcess(), &processMachine, &nativeMachine));
+        ProcessorArchitecture systemArchitecture{};
+        switch (nativeMachine)
+        {
+        case IMAGE_FILE_MACHINE_I386:
+            systemArchitecture = ProcessorArchitecture::X86;
+            break;
+        case IMAGE_FILE_MACHINE_AMD64:
+            systemArchitecture = ProcessorArchitecture::X64;
+            break;
+        case IMAGE_FILE_MACHINE_ARM64:
+            systemArchitecture = ProcessorArchitecture::Arm64;
+            break;
+        default:
+            THROW_HR(HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED));
```

Comment:
SUGGESTION: `THROW_HR_MSG(..., "nativeMachine=%hu", nativeMachine);`

In case we get an unrecognized value, log the value as just knowing the callstack that this line failed won't be as helpful